src/permissions/plugins/github/index.ts

```diff
@@ -0,0 +1,130 @@
+import { MessageBuilder } from '../../../MessageBuilder.js';
+import { RepositoryConfig, TeamConfig } from '../../types.js';
+import { Plugin } from '../Plugin.js';
+import { GITHUB_DEFAULT_BRANCH, GITHUB_DEFAULT_ENVIRONMENT } from '../../../constants.js';
+import { IS_DRY_RUN } from '../../../helpers.js';
+import { getOctokit } from '../../../octokit.js';
+import chalk from 'chalk';
+
+class GitHubPlugin implements Plugin {
+  handleRepo = async (
+    repo: RepositoryConfig,
+    teams: TeamConfig[],
+    org: string,
+    builder: MessageBuilder,
+  ) => {
+    const { properties } = repo;
+    if (!properties) return;
+
+    if (properties.type === 'ecosystem-npm-package') {
+      await this.ensureNpmEnvironment(repo, org, builder);
+    }
+  };
+
+  private ensureNpmEnvironment = async (
+    repo: RepositoryConfig,
+    org: string,
+    builder: MessageBuilder,
+  ) => {
+    const octokit = await getOctokit(org);
+
+    try {
+      // Check if the environment already exists
+      let environmentExists = false;
+      try {
+        await octokit.repos.getEnvironment({
+          owner: org,
+          repo: repo.name,
+          environment_name: GITHUB_DEFAULT_ENVIRONMENT,
+        });
+        environmentExists = true;
+      } catch (error: any) {
+        if (error.status !== 404) {
+          throw error;
+        }
+      }
+
+      if (!environmentExists) {
+        builder.addContext(
+          `:sparkles: Creating GitHub environment \`${GITHUB_DEFAULT_ENVIRONMENT}\` for repository \`${repo.name}\``,
+        );
+        console.info(
+          chalk.green('Creating GitHub environment'),
+          chalk.cyan(GITHUB_DEFAULT_ENVIRONMENT),
+          'for repository',
+          chalk.cyan(repo.name),
+        );
+
+        if (!IS_DRY_RUN) {
+          await octokit.repos.createOrUpdateEnvironment({
+            owner: org,
+            repo: repo.name,
+            environment_name: GITHUB_DEFAULT_ENVIRONMENT,
+            deployment_branch_policy: null,
+          });
+        }
+      }
+
+      const { data: policies } = await octokit.repos.listDeploymentBranchPolicies({
+        owner: org,
+        repo: repo.name,
+        environment_name: GITHUB_DEFAULT_ENVIRONMENT,
+      });
+
+      const hasDefaultBranchPolicy = policies.branch_policies?.some(
+        (policy: any) => policy.name === GITHUB_DEFAULT_BRANCH,
+      );
```

Comment:
I think this should be "strictly one and it is only main" rather than "one of these is main".

(`.length === 1 && foo[0].name === main`)

Reply:
Actually we should specifically treat it as "missing" and "unexpected".

i.e. for every policy we currently have, if it's not for the main branch, remove it. If we don't have one for the main branch, add it